v2/thirdpartyemailpassword/migration/account-creation/ep-migration-without-password-hash.mdx

```diff
@@ -0,0 +1,284 @@
+---
+id: ep-migration-without-password-hash
+title: User Creation without password hashes
+hide_title: true
+---
+
+import BackendSDKTabs from "/src/components/tabs/BackendSDKTabs"
+import TabItem from '@theme/TabItem';
+
+# User Creation without password hashes
+
+:::caution
+The recommended method for migrating users to SuperTokens is by [importing users with their password hashes](./user-creation).
+You should only use the following method if you do not have access to your user's password hashes and still have access to your previous identity provider.
+:::
+
+SuperTokens also supports the "**just in time**" user migration strategy for when password hashes cannot be exported from your legacy provider.
+
+We will need to make the following customizations to SuperTokens authentication flows to support this strategy:
+
+- **Step 1) Prevent signups from users who exist in the external provider.**
+  - To prevent loss of data during migration we block signups from users who have existing accounts with the external provider.
+- **Step 2) Create a SuperTokens account for users trying to sign in if they have an account with the external provider.**
+  -  We will modify the signin flow to check if the user signing in has an existing account with the external provider. If they do and the input credentials are valid, we create a SuperTokens user and import their user data.
+- **Step 3) Create a SuperTokens account for users who have an account with the external provider but have forgotten their password.**
+  - Some users who have an account with the external provider and not with SuperTokens may have forgotten their passwords and will trigger the password reset flow.
+   Since SuperTokens requires an account to exist to perform a password reset, we will need to modify the password reset flow to check that if the user needs to be migrated. If they do, we create a SuperTokens account with a temporary password, import their user data and continue the password reset flow.
+   - To ensure that users can only signin once they succesfully reset their passwords we add the `isUsingTemporaryPassword` flag to the accounts metadata. We will also modify the signin flow to block signins from accounts with this metadata.
+- **Step 4) Remove the `isUsingTemporaryPassword` flag on successful password reset**
+  - Once the password has been successfully reset we check if the user has the `isUsingTemporaryPassword` flag set in the metadata. If they do we will clear the flag from the user's metadata.
+
```

Comment:
Add list of edge cases from the user flow point of view at the end of this file and also link to that here.